src/components/wallet/earn/UserRewards.vue

```diff
@@ -70,6 +78,83 @@ export default class UserRewards extends Vue {
     }
 
+    get activeWallet(): WalletType {
+        return this.$store.state.activeWallet
+    }
+
+    get pendingUndepositTx(): UndepositPendingTx | null {
+        const pendingTx = this.$store.getters['Signavault/transactions'].find(
+            (item: SignavaultTx) =>
+                item?.tx?.alias === this.activeWallet.getStaticAddress('P') &&
+                WalletHelper.getUnsignedTxType(item?.tx?.unsignedTx) === 'UnlockDepositTx'
+        )
+        if (!pendingTx?.tx?.unsignedTx) {
+            return null
+        }
+
+        const unsignedTx = new UnsignedTx()
+        unsignedTx.fromBuffer(Buffer.from(pendingTx.tx.unsignedTx, 'hex'))
+
+        const tx = unsignedTx.getTransaction() as UnlockDepositTx
+
+        // Track total consumed and produced locked amounts
+        let totalConsumedLocked = new BN(0)
+        let totalProducedLocked = new BN(0)
+
+        // Extract deposit transaction IDs
+        const depositTxIDs: string[] = []
+
+        // Process inputs to find consumed locked amounts and deposit txIDs
+        const ins = tx.getIns()
+        for (let i = 0; i < ins.length; i++) {
+            const input = ins[i]
+            const baseInput = input.getInput()
+
+            if (baseInput && baseInput._typeName === 'LockedIn') {
+                // Get consumed locked amount
+                const consumedLocked = baseInput.getInput().amount
+                if (Buffer.isBuffer(consumedLocked)) {
+                    const bnAmount = bintools.fromBufferToBN(consumedLocked)
+                    totalConsumedLocked = totalConsumedLocked.add(bnAmount)
+                }
+
+                // Extract deposit txID
+                const lockedIn = baseInput as any
+                if (lockedIn.ids && lockedIn.ids.depositTxID && lockedIn.ids.depositTxID.txid) {
+                    const depositTxIDBuffer = lockedIn.ids.depositTxID.txid
+                    const depositTxID = bintools.cb58Encode(depositTxIDBuffer)
+                    if (!depositTxIDs.includes(depositTxID)) {
+                        depositTxIDs.push(depositTxID)
+                    }
+                }
+            }
+        }
+
+        // Process outputs to find produced locked amounts
+        const outs = tx.getOuts()
+        for (let i = 0; i < outs.length; i++) {
+            const output = outs[i]
+            const outputObj = output.getOutput()
+
+            if (outputObj && outputObj._typeName === 'LockedOut') {
+                const producedLocked = outputObj.getOutput().amount
+                if (Buffer.isBuffer(producedLocked)) {
+                    const bnAmount = bintools.fromBufferToBN(producedLocked)
+                    totalProducedLocked = totalProducedLocked.add(bnAmount)
+                }
+            }
+        }
+
+        // Calculate amount to undeposit (consumedLocked - producedLocked)
+        const amountToUndeposit = totalConsumedLocked.sub(totalProducedLocked)
```

Comment:
This now correctly computes the undeposit amount. We also talked in our meeting about a sanity check / safety net where the total unlocked amount must be at least the undeposit amount minus the tx fee (to be sure that no accidental burn of the undeposited amount is happening).
Is there a reason why this is not covered here?

Reply:
I apologize for overlooking the sanity check we discussed in our meeting. you're right that this safety measure is important, and I should have included it in my initial implementation.